lib/improver/cli/recursive_filter.py

```diff
@@ -31,75 +31,20 @@
-        "direction and vice versa.")
-    parser.add_argument("input_filepath", metavar="INPUT_FILE",
-                        help="A path to an input NetCDF file to be processed")
-    parser.add_argument("output_filepath", metavar="OUTPUT_FILE",
-                        help="The output path for the processed NetCDF")
-    parser.add_argument("--input_filepath_alphas_x", metavar="ALPHAS_X_FILE",
-                        help="A path to a NetCDF file describing the alpha "
-                        "factors to be used for smoothing in the x "
-                        "direction")
-    parser.add_argument("--input_filepath_alphas_y", metavar="ALPHAS_Y_FILE",
-                        help="A path to a NetCDF file describing the alpha "
-                        "factors to be used for smoothing in the y "
-                        "direction")
-    parser.add_argument("--alpha_x", metavar="ALPHA_X",
-                        default=None, type=float,
-                        help="A single alpha factor (0 < alpha_x < 1) to be "
-                        "applied to every grid square in the x "
-                        "direction.")
-    parser.add_argument("--alpha_y", metavar="ALPHA_Y",
-                        default=None, type=float,
-                        help="A single alpha factor (0 < alpha_y < 1) to be "
-                        "applied to every grid square in the y "
-                        "direction.")
-    parser.add_argument("--iterations", metavar="ITERATIONS",
-                        default=1, type=int,
-                        help="Number of times to apply the filter, default=1 "
-                        "(typically < 5)")
-    parser.add_argument('--input_mask_filepath', metavar='INPUT_MASK_FILE',
-                        help='A path to an input mask NetCDF file to be '
-                        'used to mask the input file.')
-    parser.add_argument("--re_mask", action='store_true', default=False,
-                        help="Re-apply mask to recursively filtered output.")
-
-    args = parser.parse_args(args=argv)
-
-    # Load Cubes.
-    cube = load_cube(args.input_filepath)
-    mask_cube = load_cube(args.input_mask_filepath, allow_none=True)
-    alphas_x_cube = load_cube(args.input_filepath_alphas_x, allow_none=True)
-    alphas_y_cube = load_cube(args.input_filepath_alphas_y, allow_none=True)
-    # Process Cube
-    result = process(cube, mask_cube, alphas_x_cube, alphas_y_cube,
-                     args.alpha_x, args.alpha_y, args.iterations, args.re_mask)
-    # Save Cube
-    save_netcdf(result, args.output_filepath)
-
-
-def process(cube, mask_cube=None, alphas_x_cube=None, alphas_y_cube=None,
-            alpha_x=None, alpha_y=None, iterations=1, re_mask=False):
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube,
+            *,
+            mask_cube: cli.inputcube = None,
+            alphas_x_cube: cli.inputcube = None,
+            alphas_y_cube: cli.inputcube = None,
```

Comment:
Cube loading should all be positional, but with more than one being optional I can see them problem here.
This could be solved with loading all the cubes into an array of cubes and then working out from the number of cubes which they are.
1 cube: mask
2 cubes: alphas
3 cubes mask and alphas.

@TomekTrzeciak any thoughts?

Reply:
@TomekTrzeciak and I discussed this, we have a good idea for how this should look - but probably a little hard to explain through GitHub comments. I can put up a PR to this PR if that's okay?

(We'll also need to update the ancil generating CLI, which I'll attack now as well)

Reply:
Yes that is fine, I'll be interested to see what the solution is?